android/jni/com/mapswithme/platform/Platform.cpp

```diff
@@ -164,6 +149,13 @@ namespace android
   void Platform::SetStoragePath(string const & path)
   {
     m_writableDir = path;
+
+    // @TODO it's a bug when user had all his maps on SD but when space is low,
+    // he loses access to all downloaded maps. We should display warnings in these cases in UI.
+    if (!HasAvailableSpaceForWriting(1024))
```

Comment:
Didn’t understand, how it works correctly in your scenario.
If there is not enough free space, download will fail. And If we are not saying a word about the real reason (do we?), user will not understand, why it has failed.

> On Oct 7, 2015, at 01:16, ygorshenin <notifications@github.com> wrote:
> 
> In android/jni/com/mapswithme/platform/Platform.cpp <https://github.com/mapsme/omim/pull/131#discussion_r41362659>:
> 
> > @@ -164,6 +149,13 @@ namespace android
> >    void Platform::SetStoragePath(string const & path)
> >    {
> >      m_writableDir = path;
> > +
> > +    // @TODO it's a bug when user had all his maps on SD but when space is low,
> > +    // he loses access to all downloaded maps. We should display warnings in these cases in UI.
> > +    if (!HasAvailableSpaceForWriting(1024))
> Вроде как проверки перед загрузкой в storage нет, но тогда загрузка просто завершит свою работу и удалит временные файлы. И все должно работать.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/131/files#r41362659>.
>